Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5150.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.ComponentModel;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+
+#if UITEST
+using CategoryAttribute = NUnit.Framework.CategoryAttribute;
+using Xamarin.Forms.Core.UITests;
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+
+	public static class ContentDescriptionEffectProperties
+	{
+		public static readonly BindableProperty ContentDescriptionProperty = BindableProperty.CreateAttached(
+			"ContentDescription",
+			typeof(string),
+			typeof(ContentDescriptionEffectProperties),
+			null,
+			propertyChanged: OnContentDescriptionChanged );
+
+		public static string GetContentDescription(BindableObject view)
+		{
+			return (string)view.GetValue(ContentDescriptionProperty);
+		}
+
+		public static void SetContentDescription(BindableObject view, string value)
+		{
+			view.SetValue(ContentDescriptionProperty, value);
+		}
+
+		static void OnContentDescriptionChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			System.Diagnostics.Debug.WriteLine($"Old value = {oldValue}, new value = {newValue}");
+		}
+	}
+
+	public class ContentDescriptionEffect : RoutingEffect
+	{
+		public const string EffectName = "ContentDescriptionEffect";
+
+		public ContentDescriptionEffect() : base($"{Issues.Effects.ResolutionGroupName}.{EffectName}")
+		{
+		}
+	}
+
+#if UITEST
+	[Category(UITestCategories.Button)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 5150, "AutomationProperties.Name, AutomationProperties.HelpText on Button not read by Android TalkBack", PlatformAffected.Android)]
+	public class Issue5150 : TestContentPage // or TestMasterDetailPage, etc ...
+	{
+
+		private void Configure(Button button, Label label, StackLayout layout, string buttonText, string buttonName = null, string buttonHelp = null)
+		{
+			button.Text = buttonText;
+			button.Effects.Add(new ContentDescriptionEffect());
+			button.SetValue(AutomationProperties.NameProperty, buttonName);
+			button.SetValue(AutomationProperties.HelpTextProperty, buttonHelp);
+			button.PropertyChanged += (object sender, PropertyChangedEventArgs e) => {
+				if (e.PropertyName == ContentDescriptionEffectProperties.ContentDescriptionProperty.PropertyName)
+				{
+					var element = sender as Button;
+					var desc = (string)element.GetValue(ContentDescriptionEffectProperties.ContentDescriptionProperty);
+					label.Text = desc;
```

Comment:
here you can use your static method
```suggestion
					label.Text = ContentDescriptionEffectProperties.GetContentDescription(button);
```